libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -162,6 +166,30 @@ export function setActiveDropdown(elem) {
   });
 }
 
+export const [hasActiveLink, setActiveLink, getActiveLink] = (() => {
+  let activeLinkFound;
+
+  return [
+    () => activeLinkFound,
+    (val) => { activeLinkFound = !!val; },
+    (area) => {
+      if (hasActiveLink() || !(area instanceof HTMLElement)) return null;
+      const { origin, pathname } = window.location;
+      let activeLink;
+
+      [`${origin}${pathname}`, pathname].forEach((path) => {
+        if (activeLink) return;
+        activeLink = area.querySelector(`a[href = '${path}'], a[href ^= '${path}?'], a[href ^= '${path}#']`);
+      });
+
+      if (!activeLink) return null;
```

Comment:
I don't think there's a risk here but still asking - would we need to set `activeLinkFound` as false for this scenario?